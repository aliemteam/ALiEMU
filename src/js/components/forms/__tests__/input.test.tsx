import React from 'react';
import renderer from 'react-test-renderer';

import Input from '../input';

it('should render correctly', () => {
    const tree = renderer.create(
        <>
            <Input label="Labelled Input" />
            <Input defaultValue="Unlabelled Input" />
            <Input placeholder="Placeholder Input" />
            <Input disabled />
            <Input raised label="Raised Input" />
        </>,
    );
    expect(tree).toMatchSnapshot();
});
