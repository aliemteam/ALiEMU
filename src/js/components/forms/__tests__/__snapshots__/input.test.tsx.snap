// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
Array [
  <label
    className=""
  >
    Labelled Input
    <div
      className="container"
    >
      <input
        className="input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
      />
    </div>
  </label>,
  <div
    className="container"
  >
    <input
      className="input"
      defaultValue="Unlabelled Input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
    />
  </div>,
  <div
    className="container"
  >
    <input
      className="input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Placeholder Input"
    />
  </div>,
  <div
    className="container disabled"
  >
    <input
      className="input"
      disabled={true}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
    />
  </div>,
  <label
    className=""
  >
    Raised Input
    <div
      className="container containerRaised"
    >
      <input
        className="input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
      />
    </div>
  </label>,
]
`;
